compiler/plugins/target/AMD-AIE/aie/AMDAIECreatePathFindFlows.cpp

```diff
@@ -253,21 +270,27 @@ LogicalResult runOnPacketFlow(
     Block &b = swbox.getConnections().front();
     builder.setInsertionPoint(b.getTerminator());
 
-    std::vector<bool> amselOpNeededVector(32);
+    std::vector<std::vector<bool>> amselOpNeededVector(
+        numArbiters, std::vector<bool>(numMsels, false));
     for (const auto &[physPort, masterSet] : masterSets) {
       if (tileLoc != physPort.tileLoc) continue;
-      for (int value : masterSet) amselOpNeededVector[value] = true;
+      for (auto [arbiter, msel] : masterSet)
+        amselOpNeededVector.at(arbiter).at(msel) = true;
     }
     // Create all the amsel Ops
-    DenseMap<int, AMSelOp> amselOps;
-    for (int i = 0; i < 32; i++) {
-      if (!amselOpNeededVector[i]) continue;
-      int arbiterID = i % numArbiters;
-      int msel = i / numArbiters;
-      auto amsel =
-          builder.create<AMSelOp>(builder.getUnknownLoc(), arbiterID, msel);
-      amselOps[i] = amsel;
+    DenseMap<std::pair<uint8_t, uint8_t>, AMSelOp> amselOps;
+    for (int i = 0; i < numMsels; i++) {
+      for (int a = 0; a < numArbiters; a++) {
+        if (amselOpNeededVector.at(a).at(i)) {
+          int arbiterID = a;
+          int msel = i;
+          auto amsel =
+              builder.create<AMSelOp>(builder.getUnknownLoc(), arbiterID, msel);
+          amselOps[{arbiterID, msel}] = amsel;
```

Comment:
~~nit (and I'm guilty of this too so I should fix it): `[]` copies the key so this creates`{...}`, copies, then throws away the `{...}`. You can probably~~

`DenseMap` [moves the key](https://github.com/iree-org/llvm-project/blob/9ff6c35a7e50c906537f085c0d2488ea4836dcc5/llvm/include/llvm/ADT/DenseMap.h#L363).

Reply:
`amselOps` is a `DenseMap`. Is there still an issue here?

Reply:
nah that's why i crossed out the nit